src/components/SlideElementWrapper.js

```diff
@@ -1,41 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useSpring, animated } from 'react-spring';
-
-import { SlideContext } from '../hooks/useSlide';
+import { DeckContext } from '../hooks/useDeck';
+import { AnimationMutexContext } from '../hooks/useMutex';
 
 /**
  * SlideElementWrapper provides a component for animating slideElements
  * Anything wrapped inside will be affected by the transition.
  *
  * It is currently using useSpring but ideally we will be able to switch
  * to whatever react-spring hook a user desires!
- * 
+ *
  * Note: Immediate is a React-Spring property that we pass to the animations
  * essentially it skips animations.
  */
 
 const SlideElementWrapper = ({ elementNum, transitionEffect, children }) => {
-  const [state, ,] = React.useContext(SlideContext);
+  const [
+    { reverseDirection, currentSlideElement, immediateElement }
+  ] = React.useContext(DeckContext);
+  const { signal } = React.useContext(AnimationMutexContext);
+  const activeElement = elementNum === currentSlideElement;
+  const upcomingElement =
+    (elementNum > currentSlideElement && !reverseDirection) ||
+    (elementNum < currentSlideElement && reverseDirection);
+  const previousElement =
+    (elementNum < currentSlideElement && !reverseDirection) ||
+    (elementNum > currentSlideElement && reverseDirection);
 
-  const [styleProps, set] = useSpring(() => transitionEffect.from);
+  const [styleProps, set] = useSpring(() => {
+    if ((activeElement || upcomingElement) && !previousElement) {
+      return reverseDirection ? transitionEffect.to : transitionEffect.from;
+    } else if (previousElement) {
+      return reverseDirection ? transitionEffect.from : transitionEffect.to;
```

Comment:
For previousElements, that means elements that have already animated in for forwards direction, or elements that have disappeared in reverse direction, we want to persist their completed state transitions since react-spring doesn't do that automatically.